javatests/arcs/core/host/PausableTestingHost.kt

```diff
@@ -0,0 +1,32 @@
+package arcs.core.host
+
+import arcs.core.storage.testutil.testStorageEndpointManager
+import arcs.jvm.util.testutil.FakeTime
+
+/** Fake [ArcHost] that let's us view pausing behavior for tests. */
+class PausableTestingHost(
+  schedulerProvider: SchedulerProvider,
+  vararg particles: ParticleRegistration
+) : TestingHost(
+  HandleManagerFactory(
+    schedulerProvider = schedulerProvider,
+    storageEndpointManager = testStorageEndpointManager(),
+    platformTime = FakeTime()
+  ), *particles
+) {
+
+  override suspend fun pause() {
+    super.pause()
+    numPauses++
+  }
+
+  override suspend fun unpause() {
+    super.unpause()
+    numUnpauses++
+  }
+
+  companion object {
+    var numPauses = 0
+    var numUnpauses = 0
```

Comment:
Static mutable state (e.g. as stored in companion objects) should be avoided. It can cause strange behaviour if you aren't careful to reset everything properly between runs.

I'd recommend moving these to be an instance member instead. And while you're here, I'd also suggest making this just a single `isPaused` boolean.